lib/validate.js

```diff
@@ -718,7 +718,7 @@ Vp.checkArguments = function checkArguments(ts, expected, msg, locs, loc) {
 
 // (Expression, Statement, Statement | null) -> void
 Vp.ifStatement = function ifStatement(test, cons, alt) {
-    this.checkSubtype(this.expression(test), types.boolish, "if test", test.loc);
+    this.checkSubtype(this.expression(test), ty.Int, "if test", test.loc);
```

Comment:
This fix looks good. This is a subtle area; would you mind adding a testcase in tests/index.js for this? Thanks!